hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,548 @@
+In Pass-though mode, control of the `cs_n` lines passes directly to the inter-module port, `pt_cs_ni`.
+
+### Arbitrary Length Commands
+
+Typically the length of each command is limited by the maximum number of bits in the {{< regref "COMMAND_0.TXN_CNT_0" >}}, {{< regref "COMMAND_0.RX_CNT_0" >}}, and {{< regref "COMMAND_0.TX1_CNT_0" >}} fields.
+However, a series  of SPI_HOST IP commands can be merged into one long command sequence using the {{< regref "CONFIG_0.CSAAT_0" >}} field, which leaves the `cs_n` line low after the first SPI_HOST command.
+By raising the `cs_n` line only after the last command has completed, the peripheral will interpret the resulting transmit/receive sequences as one long command.
+
+### Opportunistic Support for Back-to-back Commands
+
+There is no command queue in the SPI_HOST IP.
+However a second command can be placed into the Config/Command CDC whenever {{< regref "STATUS.READY" >}} is high, even if the previous command is still running.
+The internal state within the Config/Command CDC provides the option of scheduling a second command to execute immediately after completion of the current command.
+
+On the other hand, writing a one to {{< regref "COMMAND_0.GO_0" >}} (or any `GO` bit in this multi-register) when {{< regref "STATUS.READY" >}} is low will trigger an error condition, which must be acknowledged by software.
+
+## Data Formatting (Placeholder)
+
+**Section moved to PR #5149**
+
+## Pass-through Mode
+
+The SPI_HOST also supports a special "Pass-through" mode, which allows for the direct control of the serial interface by another block (namely SPI_DEVICE).
+
+To enable this mode, one sets {{< regref "CONTROL.PASSTHRU" >}} to one.
+
+A multiplexor then configures the output signals `cio_sck_o` and `cio_cs_no` to directly match the inter-module signals `pt_sck_i` and `pt_cs_ni[MaxCS-1:0]`.
+Likewise the bidirectional port-control `cio_sd_o[3:0]` and `cio_sdt_t[3:0]` are then respectively driven by inter-module signals `cio_sdo_i[3:0]` and `cio_sdi_o[3:0]`.
+Meanwhile the peripheral data input signal `cio_sdo_o[3:0]` is always connected to the inter-module `pt_sdi_o[3:0]`.
+
+## Interrupt aggregation
+
+In order to reduce the total number of interrupts in the system, the SPI_HOST has only two interrupt lines: `error` and `event`.
+Within these two interrupt classes, there are a number of conditions which can trigger them.
+
+Each interrupt class has a secondary status and mask register, to control which sub-classes of events will cause an interrupt.
+
+### Events and Event Interrupts
+
+The SPI_HOST supports interrupts for the following events:
+
+- `IDLE`: The SPI_HOST is idle.
+- `READY`: The SPI_HOST is ready to accept a new command.
+- `RXFULL`: The SPI_HOST has run out of room in the RXFIFO.
+- `RXSTALL`: The RXFIFO is full, and the SPI HOST is stalled and waiting for firmware to remove some data.
```

Comment:
What is the difference between these two?

Reply:
RXFULL means that the SPI HOST has exactly enough data to fill the FIFO, RXSTALL is also full, but the FSM has also stalled because it can't fetch anymore data. (it probably means the the shift register also has a byte in it that can't be grabbed.

Reply:
Is software supposed to take a different action between the two? It seems like `RXFULL` would be enough.

Reply:
Fair enough.  I've updated the text:

```
In addition to these events, there are also two additional diagnostic fields in the {{< regref "STATUS" >}} register:  
- `RXSTALL`: The RXFIFO is full, and the SPI HOST is stalled and waiting for firmware to remove some data.
- `TXSTALL`: The TX FIFO is not only empty, but the SPI HOST is stalled and waiting for firmware to add more data.

These bits can provide diagnostic data for tuning the throughput of the device, but do not themselves generate event interrupts.
```